scripts/merge_local_staging.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+import shutil
+import sys
+
+
+def copy_and_append_index(target_directory, dir_path):
+    sub_dir_name = os.path.basename(os.listdir(dir_path)[0])
```

Comment:
nit: IIRC if you use `pathlib` then you can write code like `parent / child / file_ext`, like:

https://github.com/apache/kvrocks/blob/0a301e417bf0704f440cb92d518dae74a4680c08/x.py#L228

Of course this is not a blocker.

Reply:
refer to https://github.com/apache/incubator-opendal/pull/3660#discussion_r1403842749